src/main/scala/util/RunUtils.scala

```diff
@@ -240,57 +233,43 @@ object RunUtils {
               // We want to replace all possible indirect calls based on this CFG, before regenerating it from the IR
               return
             }
             val valueSet: Map[Variable, Set[Value]] = valueSets(n)
-            val functionNames = resolveAddresses(valueSet(indirectCall.target))
-            if (functionNames.size == 1) {
+            val targetNames = resolveAddresses(valueSet(indirectCall.target)).map(_.name).toList.sorted
+            val targets = targetNames.map(name => IRProgram.procedures.filter(_.name.equals(name)).head)
+            if (targets.size == 1) {
               modified = true
-              val block = commandNode.block
-              block.jumps = block.jumps.filter(!_.equals(indirectCall))
-              block.jumps += DirectCall(
-                IRProgram.procedures.filter(_.name.equals(functionNames.head.name)).head,
-                indirectCall.condition,
-                indirectCall.returnTarget
-              )
-            } else if (functionNames.size > 1) {
+              val newCall = DirectCall(targets.head, indirectCall.condition, indirectCall.returnTarget)
+              block.jumps.remove(block.jumps.indexOf(indirectCall))
+              block.jumps.append(newCall)
+            } else if (targets.size > 1) {
               modified = true
-              functionNames.foreach(addressValue =>
-                val block = commandNode.block
-                block.jumps = block.jumps.filter(!_.equals(indirectCall))
-                if (indirectCall.condition.isDefined) {
-                  block.jumps += DirectCall(
-                    IRProgram.procedures.filter(_.name.equals(addressValue.name)).head,
-                    Option(
-                      BinaryExpr(
-                        BVAND,
-                        indirectCall.condition.get,
-                        BinaryExpr(BVEQ, indirectCall.target, addressValue.expr)
-                      )
-                    ),
-                    indirectCall.returnTarget
-                  )
-                } else {
-                  block.jumps += DirectCall(
-                    IRProgram.procedures.filter(_.name.equals(addressValue.name)).head,
-                    Option(BinaryExpr(BVEQ, indirectCall.target, addressValue.expr)),
-                    indirectCall.returnTarget
-                  )
-                }
-              )
-            } else {
-              // must be a call to R30
-              if (!indirectCall.target.equals(exitRegister)) {
-                throw new Exception(
-                  s"Indirect call ${indirectCall} has no possible targets. Value set: ${valueSet(indirectCall.target)}"
-                )
+              val procedure = c.parent.data
+              indirectCall.condition match {
+                // it doesn't seem like calls can actually have conditions in the ARM64 instruction set
+                case Some(_) => throw Exception("indirect call has a condition")
```

Comment:
If we do a path-sensitive analysis to resolve indirect calls we might want to use this.
I would also prefer to avoid throwing exceptions, or at least not `Exception`, and not handling it somewhere. Even if its only going to be thrown in case of a bad future refactoring, it makes it a pain to test if the tool just falls over.

Reply:
This is happening at the point at which we resolve indirect calls, so we wouldn't be using IndirectCall.condition anyway if we were using some sort of path-sensitive analysis.

This was an exception because it does indicate a fundamental issue, but I'll just remove calls having conditions from the IR completely now that I've confirmed it's unnecessary.

Reply:
> This is happening at the point at which we resolve indirect calls
Right of course yeah, then we can remove the indirect call condition.